### PR TITLE
fix some intra_period issues of encoder.

### PR DESCRIPTION
**Pull request by lizhong1008**

### CODE REVIEW

File: encoder/vaapiencoder_h264.cpp [hunk 1]
```diff
@@ -743,13 +743,21 @@ void VaapiEncoderH264::resetParams ()
     DEBUG("resetParams, ensureCodedBufferSize");
     ensureCodedBufferSize();
 
+    if (intraPeriod() == 0) {
+        ERROR("intra period must larger than 0");
+        assert(0);
```
Comment (xuguangxin) [inline encoder/vaapiencoder_h264.cpp]: Better return error here. assert should use for bug only. It's not user friendly if we assert on input value on core library

**Comment by xuguangxin** [comment]
Please help me better understand the issue. Where is 1024 come from, did not see any indicate here.

> enc: fix h264 core dump issue when large intra_period
> When intra_period larger than 1024, no reference list for P frame.
> It is due to poc calculated error.
> And HEVC encoding should also be fixed though no core dump found,
> because hevc poc is equal to m_frameIdx, but h264 is equal to
> 2*m_frameIdx.
> 
> Signed-off-by: Zhong Li zhong.li@intel.com
> Change-Id: I2b4625a7021fa26e1f4ee54f661fd39c155b7776

File: encoder/vaapiencoder_h264.cpp (hunk 1 above)
Comment (lizhong1008) [inline encoder/vaapiencoder_h264.cpp]: Here resetParams is a void function, if we change it to bool, upper-layer functions need to check it. And I think it's normal to assert parameters with ERROR info printed. We can see this way everywhere in  yami project and libva project.

**Comment by lizhong1008** [comment]
Hi Guangxin:
    you can check the fuction h264_get_log2_max_frame_num in vaapiencoder_h264.cpp. The max value of m_maxFrameNum is 1024.

File: encoder/vaapiencoder_h264.cpp (hunk 1 above)
Comment (xuguangxin) [inline encoder/vaapiencoder_h264.cpp]: Could we silent change this to other value? We can assert on exception. But assert on user input value maybe not so good especially when it in core library. U.Arite have this concern too.https://github.com/01org/libyami/issues/439

**Comment by xuguangxin** [comment]
+1

Hi @wudping :
Please help to verify it.
two things should be make sure.
1. we have this issue when you did not apply this patch.
2. this patch fix the issue, and did not bring side effect to normal stream. 
Please use analysiser to check the encoded file

**Comment by wudping** [comment]
ok, Guangxin

**Comment by wudping** [comment]
verify + 1

**Comment by xuguangxin** [comment]
merged as 0e4e54f1369978452b283e4c11d5ae227443d27f

**Pull request closed without merging**